IANotificationsTestProject/In App Notifications/NoticeController.swift

```diff
@@ -0,0 +1,65 @@
+import UIKit
+
+class NoticeController {
+    private var notices: [Notice] = []
+    private let noticePresenter = NoticePresenter()
+
+    private var activeViewIsBeingTouched: Bool = false
+
+    func present(_ notice: Notice) {
+        if noticePresenter.isPresenting {
+            notices.append(notice)
+        } else {
+            let noticeView = makeNoticeView(from: notice)
+
+            noticePresenter.presentNoticeView(noticeView) {
+                self.dismiss(noticeView)
+            }
+        }
+    }
+
+    func dismiss(_ noticeView: NoticeView) {
+        guard !activeViewIsBeingTouched else {
```

Comment:
With this mechanism the Notice will be dismissed, regardles if it's being touched or not right?

Perhaps when the `pressed` event happens any pending OPs could be dismissed, and re-scheduled when the Tap ends?

Reply:
I like that much better